### ISSUE TITLE
PostCSS isn't just for "~FuTuRe CsS SyNtAx~"

### ISSUE THREAD

**Issue by corysimmons** [HugoGiraudel/sass-guidelines#329]
> You can think of postprocessors as a polyfill for unsupported CSS features. For instance, you would write variables as they are described in the CSS specifications, then compile your stylesheets with a postprocessor only to find every variable occurrence gets replaced with its value, as Sass would do.

> The idea behind postprocessors is that once browsers support new features (e.g. CSS variables), the postprocessor does not compile them anymore and lets browsers take over.

> While providing tomorrow’s syntax today is something of a noble idea, I have to say I still prefer using Sass for most tasks. However, there are some occasions where I believe postprocessors are more suited than Sass and the like - CSS prefixing for instance - but we’ll get back to this.

Dude Hugo... I love ya man, but you should really learn PostCSS before posting things like this.

The big difference isn't that PostCSS lets you use future CSS syntax (that's CSSnext), it's that PostCSS gives you direct access to the power of JavaScript.

Remember that exponent function you had to make for Sass and how slow it was? PostCSS is just JS so it's literally just `Math.pow(3, 2)`.

This is a **huge** difference as it allows library creators to do **whatever they want** as opposed to being tied down by the handful of language features preprocessors offer.

In day-to-day use, Sass is nice for syntax/variables/simple mixins, and Sass can be used alongside PostCSS, but there are an insane amount of things Sass can't do, and will never be able to do, that PostCSS can do with ease...

Please update the homepage to reflect this.

Also, please join the dark side of PostCSS. We could use someone like you in the community! :dancer:

**Comment by HugoGiraudel** [comment]
Hey Cory. Take it easy my friend, it’s a document on Sass, not on PostCSS. You make it sounds like it makes the whole thing irrelevant. :)

However I definitely agree with you that Sass Guidelines should not spread misinformation on PostCSS. This section has been rewritten a few times already, and I’m happy to come up with something better to explain the main differences between Sass and PostCSS. We could bundle this in version 1.3 that should come up anytime soon.

Would you like to help me with this? I’d be happy to give you credits in the document for your contribution.

**Comment by corysimmons** [comment]
Sorry, I didn't mean to sound upset. =)

> However I definitely agree with you that Sass Guidelines should not spread misinformation on PostCSS. 

:+1:

I'm happy to help rewrite that entire section.

I'd also like to argue that Stylus is more powerful than Sass, but the argument should be that Stylus is much less popular (way less libs and ability for a team to integrate it) and it's pretty much dead nowadays. It also lets you write you code in a billion different ways which is gross.

Anyway, here's how I'd rewrite that PostCSS section:

---

Preprocessors aside, we should also mention tools like PostCSS, which has received significant exposure these last few months.

PostCSS is commonly/incorrectly referred to as a "post processor". The assumption, combined with the unfortunate name, is that PostCSS parses over CSS that has already been processed by a preprocessor. While it can work this way, it's not a requirement, so PostCSS is actually just a "processor".

PostCSS lets you access things like selectors, properties, values, etc.; process these with JavaScript; and compile those results to CSS.

For example, Autoprefixer is built with PostCSS. It takes an input.css; parses over every rule to see if vendor prefixes are needed by referencing caniuse.com; then removes and adds vendor prefixes that are needed.

This is incredibly powerful and great for building libraries that work with any preprocessor (and vanilla CSS), but PostCSS isn't particularly easy to use yet. You have to be pretty good at JavaScript to build anything with it, and its API can be confusing at times.

Sass only provides a few language-like features like `@if`/`@else`. PostCSS provides direct access to the CSS AST and JavaScript.

**TL;DR** Sass is easy and will solve most of your problems. PostCSS is difficult (if you aren't great at JavaScript) but incredibly powerful. There's no reason why you can't/shouldn't use both. In fact, PostCSS offers an official [SCSS parser](https://github.com/postcss/postcss-scss) for just this thing.

**Comment by HugoGiraudel** [comment]
Done with https://github.com/HugoGiraudel/sass-guidelines/commit/a67d1dfdf002e0c4f8c80e6affea6f7a92e351ef. Care to review @corysimmons?

**Comment by corysimmons** [comment]
Looks great :+1: :100: 

You don't have to add me to the credits or anything though. =)

**Comment by corysimmons** [comment]
Also, you should totally switch from Sass to PostCSS, or at least start thoroughly looking into it and make your own decision. It's pretty badass/fun. =)

**Comment by HugoGiraudel** [comment]
Thanks. Closing this now. :)
